src/index/base.cpp

```diff
@@ -44,9 +44,15 @@ CBlockLocator GetLocator(interfaces::Chain& chain, const uint256& block_hash)
     return locator;
 }
 
-BaseIndex::DB::DB(const fs::path& path, size_t n_cache_size, bool f_memory, bool f_wipe, bool f_obfuscate) :
-    CDBWrapper(path, n_cache_size, f_memory, f_wipe, f_obfuscate)
-{}
+BaseIndex::DB::DB(const fs::path& path, size_t n_cache_size, bool f_memory, bool f_wipe, bool f_obfuscate)
+    : CDBWrapper{{
+          .db_path = path,
+          .cache_size = n_cache_size,
+          .in_memory = f_memory,
+          .wipe_existing = f_wipe,
+          .obfuscate_data = f_obfuscate,
+          .do_compact = gArgs.GetBoolArg("-forcecompactdb", false),
```

Comment:
In commit "Use CDBWrapper::Options ctor for non-test callers" (c5566d4899d75d9030901092ecf82c97e277d6c2)

Would be good not to add news gArgs calls here and below